driver/src/contracts/stablex_contract.rs

```diff
@@ -192,24 +178,18 @@ fn encode_prices_for_contract(price_map: &HashMap<u16, u128>) -> (Vec<u128>, Vec
 }
 
 fn encode_execution_for_contract(
-    orders: Vec<Order>,
-    executed_buy_amounts: Vec<u128>,
+    executed_orders: &[ExecutedOrder],
 ) -> (Vec<Address>, Vec<u16>, Vec<u128>) {
-    assert_eq!(
-        orders.len(),
-        executed_buy_amounts.len(),
-        "Received inconsistent auction result data."
-    );
     let mut owners = vec![];
     let mut order_ids = vec![];
     let mut volumes = vec![];
-    for (order_index, buy_amount) in executed_buy_amounts.into_iter().enumerate() {
-        if buy_amount > 0 {
+    for order in executed_orders {
+        if order.buy_amount > 0 {
             // order was touched!
             // Note that above condition is only holds for sell orders.
```

Comment:
Does this comment still make sense? Can it be possible (even before this change) that an order has only one of sell amount and buy amount equal to zero but not the other?